### PR TITLE
fix: redirect non-member users

### PR DESCRIPTION
**Pull request by joanamcsa**
**What type of PR is this?**

Fix: A bug fix

**What this PR does / why we need it:**

When accessing a room page, a verification is done server side to prevent non-members of that room to view that page.

**Which issue(s) this PR fixes:**

Fixes [#50](https://github.com/xgeekshq/intelli-mate/issues/50)

**Special notes for your reviewer:**

https://github.com/xgeekshq/intelli-mate/assets/97113028/16c0321b-c444-4326-95e1-b7be6b008801

### CODE REVIEW

File: packages/web-ui/app/rooms/[room]/layout.tsx [hunk 1]
```diff
@@ -35,6 +41,11 @@ export default async function RoomLayout({
   params: { room: string };
 }) {
   const room = await getRoom(params.room);
+
+  if (!room) {
+    redirect('/');
+  }
+
```
Comment (comoser) [inline packages/web-ui/app/rooms/[room]/layout.tsx:48]: Can't this be done directly in line 27?

File: packages/web-ui/components/chat.tsx [hunk 2]
```diff
@@ -201,12 +206,13 @@ export default function Chat({ roomId }: ChatProps) {
       data: createSocketMessageRequestFactory(roomId, value, userId ?? ''),
     });
   };
+
   return (
     <div className="flex h-full w-full flex-col">
       <ScrollArea ref={bottomEl} className="h-full">
         <div className="w-full px-4 pt-4">
           {messages.map((message, index) => {
```
Comment (comoser) [inline packages/web-ui/components/chat.tsx:214]: well done, just forgot to remove the `index`

**Review by comoser — APPROVED** [review]

File: packages/web-ui/app/rooms/[room]/layout.tsx (hunk 1 above)
Comment (joanamcsa) [reply packages/web-ui/app/rooms/[room]/layout.tsx:48]: I tried, but the `redirect` fails and the page is rendered anyway

image

File: packages/web-ui/components/chat.tsx (hunk 2 above)
Comment (joanamcsa) [reply packages/web-ui/components/chat.tsx]: yup, done here 169548d6ca96e1254caff94dd27eda4cb7a7e302

**Pull request merged**